### ISSUE TITLE
State's datetime string is modified to UTC by checkpoint for no reason

### ISSUE THREAD

**Issue by stktung** [EventStore/EventStore#2065]
**Describe the bug**
TLDR: A projection state's datetime string is incorrectly updated to UTC timezone after some checkpoints are made. The projection retrieves events from a category.

I've stumbled upon an extremely weird issue here with states in projection.

I have a continuous projection that captures a datetime string from a category in its state. For example:
```
{
  "MyTime": "2019-11-21T11:11:50+08:00"
}
```
All the datetimes from my stream are stored in +08 time zone.

For some weird reason, after a checkpoint is made (or when the projection is disabled), the state's datetime string timezone is converted to UTC +00:
```
{
  "MyTime": "2019-11-21T03:11:50+00:00"
}
```
Notice how it also converts the actual hour as well as the timezone.

This baffles me because we're just dealing with strings here .. I can't imaging the projection engine converting all datetime strings to UTC by itself!

My projection:
```
fromCategory('testdatestream')
.foreachStream()
.when({
    $init: function() {
        return {
        };
    },
    TimeChanged: function(state, event) {
        state.MyTime = event.body.MyTime;
    }
});
```
**To Reproduce**
1. Start docker container
```
docker pull eventstore/eventstore
docker volume create testtimeeventstore-data
docker volume create testtimeeventstore-log
docker run --name testtime-eventstore -it -p 2113:2113 -p 1113:1113 --mount source=testtime-eventstore-data,target=/var/lib/eventstore --mount source=testtime-eventstore-log,target=/var/log/eventstore eventstore/eventstore
```

2. Create 20,000 events with its own unique stream in the SAME category e.g.
```
testdatestream-1
testdatestream-2
...
testdatestream-20000
```
each event body only has unique timestamp in +08 timezone:
```
{
  "MyTime": "2019-11-21T10:58:04+08:00"
}
```
I have a handy powershell script here that does it:
```
(1..20000) | %{ Invoke-WebRequest -Uri ("http://localhost:2113/streams/testdatestream-1") -Headers @{ "Content-Type" = "application/json"; "ES-EventType" = "TimeChanged"; "ES-EventId" = [System.Guid]::NewGuid().ToString(); } -Body ("{ `"MyTime`": `"" + ([DateTime]::UtcNow.AddSeconds($_).ToLocalTime().ToString("yyyy-MM-ddThh:mm:sszzz"))  + "`" }") -Method Post }
```

3. Create the projection (see above projection snippet)

4. Start the projection and check the state for testdatestream-1. *Notice the datetime string is in +08 time zone.*

![image](https://user-images.githubusercontent.com/132192/69394267-84ff8000-0d16-11ea-9d71-e1b244db1f5c.png)

5. Wait for a few checkpoints to complete.

**Expected behavior**
The state should not be changed at all.

**Actual behavior**
After a few checkpoints have past, the state's *datetime string is now UPDATED to UTC +00 timezone!* (remember there's only one event per stream in the category)

![image](https://user-images.githubusercontent.com/132192/69394326-b24c2e00-0d16-11ea-90ec-158b2c2df11c.png)

**EventStore details**
 - EventStore server version: 5.0.1, 5.0.5, 4.1.3

**Additional context**
- My computer's time is in the +08 timezone (Hong Kong).

**Comment by pgermishuys** [comment]
Thanks for raising this issue @stktung and for the great reproduction. We have identified the root cause and will address it.

**Comment by Faq** [comment]
@pgermishuys this should be closed, thnx

**Comment by pgermishuys** [comment]
Thanks for the heads up @Faq :)
